src/System.Slices/System/Buffers/Memory.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Represents a Span<byte> factory</byte>
+    /// </summary>
+    /// <remarks>
+    /// This struct is not safe for multithreaded access, i.e. it's subject to struct tearing that can result in unsafe memory access. 
```

Comment:
@jkotas fair but if `Memory<T>` and `Span<T>` map 1-1 right now so my naive POV is that it needs to be as cheap as a `Span<T>` or the usage pattern needs to be different. I'm having a hard time coming with an alternative plan way to avoid slicing the `Memory<T>` to show a sliding window of the full memory that is currently writable.